onnx/defs/quantization/defs.cc

```diff
@@ -158,61 +158,136 @@ y_zero_point = cast(round(saturate(itermediate_zero_point)))
 ```
 
 * where qmax and qmin are max and min values for quantization range .i.e [0, 255] in case of uint8
-* for saturation, it saturates to [0, 255] if it's uint8, or [-127, 127] if it's int8. Right now only uint8 is supported.
+* for saturation, it saturates to [0, 255] if it's uint8, or [-127, 127] if it's int8, or [-f8_max, f8_max]
+  for any float 8 types
 * rounding to nearest ties to even.
 
 Data quantization formula is:
 ```
 y = saturate (round (x / y_scale) + y_zero_point)
 ```
 
-* for saturation, it saturates to [0, 255] if it's uint8, or [-127, 127] if it's int8. Right now only uint8 is supported.
-* rounding to nearest ties to even.
+y_zero_point must be 0 for any float 8 type.
```

Comment:
But this is a problem. If we are forced to use 0 as the zeropoint, the above computation of scale will not guarantee that all values can be reasonably represented as a float8. We may need to change the computation of scale as well, using something like "max ( max(x)/qmax, min(x)/qmin )" with some adjustments for rounding etc.

But, better still, if this is being used in practice, what are they doing?

Reply:
May need some adjustments to ensure signs are handled correctly as well.